elogia_sale/models/elogia_sale.py

```diff
@@ -226,12 +231,86 @@ def calc_amount_total(self):
                     val_tmp = record.invoice_provision / record.currency_id.rate
                 else:
                     val_tmp = (record.invoice_provision / record.currency_id.rate) * record.control_id.currency_id.rate
+                    if record.currency_id == record.control_id.company_id.currency_id:
+                        val_tmp = record.invoice_provision * record.control_id.currency_id.rate
             record.invoice_provision_ml = val_tmp
 
-    @api.model
     def create_purchase_order(self):
-        for record in self:
-            pass
+        list_supplier = [line for line in self if line.state == 'no_process' and line.type_payment != 'client'
+                         and line.control_id.state in ['pending', 'sale']]
```

Comment:
Lo cambiaría por:
list_supplier =  self.filtered(lambda a: a.state ==  'no_process' and a.type_payment != 'client' and a.control_id.state in ['pending', 'sale'])

Reply:
Oído